cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -284,7 +279,16 @@ impl ABIMachineSpec for AArch64MachineDeps {
                     RegClass::Vector => unreachable!(),
                 };
 
-                if *next_reg < max_per_class_reg_vals && remaining_reg_vals > 0 {
+                let push_to_reg =
+                    if call_conv == isa::CallConv::Winch && args_or_rets == ArgsOrRets::Rets {
+                        // Winch uses the first registry to return the last result
+                        i == params.len() - 1
+                    } else {
+                        // Use max_per_class_reg_vals & remaining_reg_vals otherwise
+                        *next_reg < max_per_class_reg_vals && remaining_reg_vals > 0
+                    };
```

Comment:
Rather than checking for winch returns every iteration of the loop, can we initialize `remaining_reg_vals` such that this behavior happens automatically? I think that means initializing it to one when we are doing winch returns. That feels cleaner to me than adding edge case conditions to check in the middle of the body of the register assignment loop.

Reply:
If `remaining_reg_vals` is one, it will put the first result in the registry, not the last one. I can add another variable named `skip_reg_vals` which is zero except when processing winch returns, then it is `params.len() - 1`. What do you think ?

Reply:
Ah I missed that detail. In that case, I think it is fine as-is.